javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,563 @@
+        }
+
+        @Override
+        public int lastIndexOf(Object obj) {
+            return Seq.<Object>narrow(delegate).lastIndexOf(obj);
+        }
+
+        @Override
+        public ListIterator<T> listIterator() {
+            return delegate.toJavaList().listIterator();
+        }
+
+        @Override
+        public ListIterator<T> listIterator(int index) {
+            return delegate.toJavaList().listIterator(index);
+        }
+
+        @Override
+        public T remove(int index) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.remove(int)
+            return setDelegateAndGetPreviousElement(index, delegate.removeAt(index));
+        }
+
+        @Override
+        public boolean remove(Object o) {
+            // may throw a ClassCastException accordingly to the j.u.List.remove(Object)
+            @SuppressWarnings("unchecked") final T that = (T) o;
+            return setDelegateAndCheckChanged(delegate.remove(that));
+        }
+
+        @Override
+        public boolean removeAll(Collection<?> c) {
+            // may throw a ClassCastException accordingly to the j.u.List.removeAll(Collection)
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) c;
+            return setDelegateAndCheckChanged(delegate.removeAll(that));
+        }
+
+        @Override
+        public boolean retainAll(Collection<?> c) {
+            // may throw a ClassCastException accordingly to j.u.List.retainAll(Collection)
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) c;
+            return setDelegateAndCheckChanged(delegate.retainAll(that));
+        }
+
+        @Override
+        public T set(int index, T element) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.set(int, T)
+            return setDelegateAndGetPreviousElement(index, delegate.update(index, element));
+        }
+
+        @Override
+        public int size() {
+            return delegate.size();
+        }
+
+        @Override
+        public List<T> subList(int fromIndex, int toIndex) {
+            // may throw IndexOutOfBoundsException accordingly to j.u.List.subList(int, int)
+            return delegate.subSequence(fromIndex, toIndex).asJava();
+        }
+
+        @Override
+        public Object[] toArray() {
+            return delegate.toJavaArray();
+        }
+
+        @Override
+        public <U> U[] toArray(U[] array) {
+            return delegate.toJavaList().toArray(array);
+        }
+
+        // -- Object.*
+
+        @Override
+        public boolean equals(Object o) {
+            return (o == this) || ((o instanceof java.util.List) && areEqual(delegate, (java.util.List<?>) o));
+        }
+
+        @Override
+        public int hashCode() {
+            return delegate.hashCode();
+        }
+
+        @Override
+        public String toString() {
+            return delegate.toString();
```

Comment:
I'm still unsure what to return but not the string representation of the delegate because this is of type j.u.List. Will think about it.